Google.PowerShell/BigQuery/BqTabledata.cs

```diff
@@ -431,4 +431,76 @@ protected override void ProcessRecord()
             }
         }
     }
+
+    /// <summary>
+    /// <para type="synopsis">
+    /// Retrieves table data from a specified set of rows.
+    /// </para>
+    /// <para type="description">
+    /// Retrieves table data from a specified set of rows. Requires the READER dataset role.  
+    /// You can specify how many results to return per page.  This will affect how fast the 
+    /// data is returned.  Rows are returned as Google.Cloud.BigQuery.V2.BigQueryRow objects.
+    /// Data can be extracted by indexing by column name (ex: (string) row["title"]; ).
+    /// </para>
+    /// <example>
+    ///   <code>
+    /// PS C:\> $table = get-bqtable -DatasetID "book_data" "classics"
+    /// PS C:\> $list = $table | get-bqtabledata
+    ///   </code>
+    ///   <para>Fetches all of the rows in book_data:classics and exports them to $list</para>
+    ///   <code>
+    /// PS C:\> $list = $table | Get-BqTabledata -PageSize 100
+    ///   </code>
+    ///   <para>This does the same as above, but fetches them in batches of 100.  This means 
+    ///   if the table is larger, the results will arrive faster, but will take more time 
+    ///   overall due to additional calls to the server.</para>
+    /// </example> 
+    /// <para type="link" uri="(https://cloud.google.com/bigquery/docs/reference/rest/v2/tabledata)">
+    /// [BigQuery Tabledata]
+    /// </para>
+    /// </summary>
+    [Cmdlet(VerbsCommon.Get, "BqTabledata")]
+    public class GetBqTabledata : BqCmdlet
+    {
+        /// <summary>
+        /// <para type="description">
+        /// The table to export rows from.
+        /// </para>
+        /// </summary>
+        [Parameter(Mandatory = true, ValueFromPipeline = true, Position = 0)]
+        [ValidateNotNull]
+        [PropertyByTypeTransformation(TypeToTransform = typeof(Table), Property = nameof(Table.TableReference))]
+        public TableReference InputObject { get; set; }
+
+        /// <summary>
+        /// <para type="description">
+        /// The number of results returned per call to the server. If there are more results
+        /// than the initial page, they will be fetched lazily with subsequent pages. 
+        /// </para>
+        /// </summary>
+        [Parameter(Mandatory = false)]
+        public int? PageSize { get; set; }
```

Comment:
I agree. It is an implementation detail we should hide from the user. We should just "do the right thing".

The only scenario I can come up with where the user would want it is if they know exactly how many rows they want to inspect, and don't want to waste data cap getting extra. I don't think that is a scenario we need to cater to.